### ISSUE TITLE
R-Pi 2: Syntax Error using v4.2.4 of node.js 

### ISSUE THREAD

**Issue by mauijim68** [jishi/node-sonos-http-api#128]
Hi all-

I tried to install node-sones-api on my Raspberry Pi 2 tonight.  I ran update and upgrade to be sure the Pi was updated to its latest releases.  I took the binary from node.org for Arm v71 v4.2.4 of node.js and installed in into /usr/local (since installing using apt-get gets you the .12 release). 

However, I get an Syntax error after running npm start.  Any ideas?  The files looked ok, so I assume I must be missing something in my setup.

TIA!

```
pi@SunSpot ~/node-sonos-http-api-master $ npm install package
package@1.0.1 node_modules/package
pi@SunSpot ~/node-sonos-http-api-master $ npm start

> sonos-http-api@0.17.0 start /home/pi/node-sonos-http-api-master
> node server.js

no settings file found, will only use default settings
/home/pi/node-sonos-http-api-master/lib/actions/._clearqueue.js:1
(function (exports, require, module, __filename, __dirname) { 
                                                              

SyntaxError: Unexpected token ILLEGAL
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:414:25)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at requirePath (/home/pi/node-sonos-http-api-master/node_modules/require-fu/lib/index.js:14:22)
    at /home/pi/node-sonos-http-api-master/node_modules/require-fu/lib/index.js:21:9
    at Array.forEach (native)
    at fuDirectory (/home/pi/node-sonos-http-api-master/node_modules/require-fu/lib/index.js:20:34)
    at /home/pi/node-sonos-http-api-master/node_modules/require-fu/lib/index.js:25:5
    at new HttpAPI (/home/pi/node-sonos-http-api-master/lib/sonos-http-api.js:36:24)
    at Object.<anonymous> (/home/pi/node-sonos-http-api-master/server.js:40:11)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)

npm ERR! Linux 4.1.13-v7+
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v4.2.4
npm ERR! npm  v2.14.12
npm ERR! code ELIFECYCLE
npm ERR! sonos-http-api@0.17.0 start: `node server.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sonos-http-api@0.17.0 start script 'node server.js'.
npm ERR! This is most likely a problem with the sonos-http-api package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node server.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls sonos-http-api
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/pi/node-sonos-http-api-master/npm-debug.log
pi@SunSpot ~/node-sonos-http-api-master $ 
```

**Comment by jishi** [comment]
Well, this is weird... if you remove the clearqueue.js file, does it break on the next one instead? Can't see something offending in that file, and it is probably the first one loaded.

However, I'm not sure why you ran `npm install package`, since that would also install a new npm module called "package". You should only run `npm install --production`(production indicates that it only installs the dependencies required to run, not anything related to development). 

If you want to try again, remove the node_modules folder and run `npm install --production`agian, just to be sure.

**Comment by mauijim68** [comment]
Ok, I found the issue!

It seems that when I downloaded the file with MacOS (the zip) then transferred it to the Pi, OS X decided to put a bunch of odd binaries (one for each .js file) in the directories.  It was these files that cause node to barf.

I removed the entire directory, wget the tar file directly from github and it worked fine on both Mac and Pi.

Thanks.  That was unexpected and not so obvious until I pulled the actual source file but it still choked.  (I thought the ._clearqueue.js was actually an artifact from node).

Thanks!
